packages/api/websocket/src/middleware/auth.ts

```
@@ -0,0 +1,391 @@
+ * @param socket - Socket.IO socket instance
+ * @param next - Next middleware function
+ */
+export function authMiddleware(socket: ExtendedSocket, next: (err?: Error) => void): void {
+  // Extract client IP for rate limiting
+  const clientIP = socket.handshake.address;
+  const userAgent = socket.handshake.headers['user-agent'] || 'unknown';
+  const identifier = `${clientIP}:${userAgent}`;
+
+  // Check rate limiting
+  if (!checkRateLimit(identifier)) {
+    return next(new AuthenticationError('Rate limit exceeded. Please try again later.'));
```

Biome (2.1.2)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)